applications/crossbar/src/modules/cb_storage.erl

```diff
@@ -412,3 +414,72 @@ doc_id({'user', UserId, _AccountId}) -> UserId;
 doc_id({'reseller_plans', _AccountId}) -> 'undefined';
 doc_id({'reseller_plan', PlanId, _AccountId}) -> PlanId;
 doc_id(Context) -> doc_id(scope(Context)).
+
+-spec maybe_check_storage_settings(cb_context:context(), kz_term:ne_binary()) ->
+                                          cb_context:context().
+maybe_check_storage_settings(Context, ReqVerb) when ReqVerb =:= ?HTTP_PUT
+                                                    orelse ReqVerb =:= ?HTTP_POST
+                                                    orelse ReqVerb =:= ?HTTP_PATCH ->
+    case cb_context:resp_status(Context) of
+        'success' ->
+            lager:debug("Validating storage settings"),
+            Doc = cb_context:doc(Context),
+            DbName = kz_util:format_account_db(cb_context:account_id(Context)),
+            DocId = doc_id(Context),
+            %% Files content will differ at least on this value and also this `Random'
+            %% value is used to make sure we always send unique attachment names when
+            %% testing storage settings.
+            Random = kz_binary:rand_hex(16),
+            Content = <<"some random content: ", Random/binary>>,
+            AName = <<Random/binary, "_test_credentials_file.txt">>,
+            kz_json:foldl(
+              fun(AttId, Att, ContextAcc) ->
```

Comment:
move this to a separate function. you also need to check if validation is needed (did it change ?)